### PR TITLE
[ISSUE-4264] Moving plugins.forEach { it.onUserSet(user) } after plugins are created

### PR DESCRIPTION
**Pull request by leandroBorgesFerreira**
### 🎯 Goal

Make events work again in the SDK.

### 🛠 Implementation details

- Moving `plugins.forEach { it.onUserSet(user) }`  in a later stage of the SDK initialization. 

### 🎨 UI Changes

<table>
<thead>
<tr>
<th>Before</th>
<th>After</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<video src="https://user-images.githubusercontent.com/10619102/194939491-18dd6580-2a0e-40a8-85ce-a54262aaf833.mov" controls="controls" muted="muted" />
</td>
<td>
<video src="https://user-images.githubusercontent.com/10619102/194939527-c6dc9bbd-2253-4a72-84a4-d895ef3974aa.mov" controls="controls" muted="muted" />
</td>
</tr>
</tbody>
</table>

### 🧪 Testing

Send messages to the sample app and check that they are arriving. 

#### General
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Assigned a person / code owner group (required)
- [x] Thread with the PR link started in a respective Slack channel (#android-chat-core or #android-chat-ui) (required)
- [x] PR targets the `develop` branch
- [x] PR is linked to the GitHub issue it resolves

#### Code & documentation
- ~[ ] Changelog is updated with client-facing changes~
- [x] New code is covered by unit tests
- [x] Comparison screenshots added for visual changes
- ~[ ] Affected documentation updated (KDocs, docusaurus, tutorial)~

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### 🎉 GIF
giphy (2)
_Ops D=_

### CODE REVIEW

File: stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt [hunk 1]
```diff
@@ -269,7 +269,7 @@ internal constructor(
      * @see [Plugin]
      */
     @PublishedApi
-    internal var plugins: List<Plugin> = emptyList()
+    internal lateinit var plugins: List<Plugin>
```
Comment (bychkovdmitry) [inline stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt:272]: What if somebody calls an SDK method without setting the user? Will the error be better than not initialized property like with the line above?

Comment (leandroBorgesFerreira) [reply stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt:272]: I would say that an error is better than an inconsistent state. But I think we should write a better message than "property not initialized"

Comment (bychkovdmitry) [reply stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt:272]: I checked, and when I call `createChannel` on the login screen the app crashes.
```
ChatClient.instance().createChannel("abc", "abc", emptyList(), emptyMap()).enqueue()
```

Comment (leandroBorgesFerreira) [reply stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt]: Ok, let's keep with the initial state.

**Review by kanat — APPROVED** [review]
LGTM

**Pull request merged**